### PR TITLE
feat(shared/functions): shared function unification

### PR DESCRIPTION
**Pull request by ArturMichalak**
## Description

unifies the mechanism of the shared functions and adds some additional

## Checklist

- [ ] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [X] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: shared/functions.lua [hunk 1]
```diff
@@ -10,57 +10,68 @@ function public.getIsCloseToCoords(coord1, coord2, distance)
...
-local function getIsVehicleOnList(vehicle, list)
-    local vehicleHash = GetEntityModel(vehicle)
+local function getIsOnList(item, list)
     for i = 1, #list do
-        if vehicleHash == joaat(list[i]) then
+        if item == list[i] then
             return true
         end
     end
 end
```
Comment (Manason) [inline shared/functions.lua:19]: Replace with https://overextended.dev/ox_lib/Modules/Table/Shared#libtablecontains so this function can be deleted.

**Review by Manason — APPROVED** [review]

File: shared/functions.lua (hunk 1 above)
Comment (ArturMichalak) [reply shared/functions.lua:19]: I'm not sure it's the best option when we care about performance
https://github.com/overextended/ox_lib/blob/7ecd7dc21fe5f554a0bb49f058063395d9d03402/imports/table/shared.lua#L7-L30

Comment (Manason) [reply shared/functions.lua:19]: Because of the pairs vs iterative? Fine to keep this then.

Comment (ArturMichalak) [reply shared/functions.lua:19]: exactly

**Pull request merged**